internal/core/component.rs

```diff
@@ -32,17 +43,35 @@ pub struct ComponentVTable {
         index: usize,
     ) -> core::pin::Pin<VRef<ItemVTable>>,
 
+    /// Return the range of indices below the dynamic `ItemTreeNode` at `index`
+    pub get_subtree_range:
+        extern "C" fn(core::pin::Pin<VRef<ComponentVTable>>, index: usize) -> IndexRange,
+
+    /// Return the `ComponentRc` at `subindex` below the dynamic `ItemTreeNode` at `index`
+    pub get_subtree_component: extern "C" fn(
+        core::pin::Pin<VRef<ComponentVTable>>,
+        index: usize,
+        subindex: usize,
+        result: &mut vtable::VWeak<ComponentVTable, Dyn>,
+    ),
+
     /// Return the item tree that is defined by this `Component`.
     /// The return value is an item weak because it can be null if there is no parent.
     /// And the return value is passed by &mut because ItemWeak has a destructor
     pub get_item_tree: extern "C" fn(core::pin::Pin<VRef<ComponentVTable>>) -> Slice<ItemTreeNode>,
 
+    // FIXME: This does return an invalid ItemWeak now that points to the parent repeater!
```

Comment:
Maybe we should just rename this to `parent_node`.  Maybe also rename Item{Rc,Weak} to something like Node{Rc,Weak}.    or ItemNodeRc ?

Reply:
I do not want to rename ItemRc and co.: Those are really nice to have at the logical (Component-spanning) Item tree!

The Component is below that level though: It only covers the physical Itemtree with all the repeaters and stuff... at this level we will end up with thigs that are either an Item or a Subtree. It would be cool if we could keep those two levels separate on the type system level.